101-mongodb-sharded-on-centos/scripts/config_primary.sh

```diff
@@ -0,0 +1,192 @@
+#!/bin/bash
+
+mongoAdminUser=$1
+mongoAdminPasswd=$2
+location=$3
+dnsNamePrefix=$4
+
+GetMongoPackage() {
+	packageUri=$1
+	
+	while true; do
+	#install mongo package
+    wget $packageUri && break || {
+      if [[ $n -lt 3 ]]; then
+        ((n++))
+        echo "Command failed. Attempt $n of 3:"
+        sleep 15;
+      else
+        echo "Failed to get the package $packageUri after $n attempts."
+		exit 1
+      fi
+    }
+  	done
+}
+
+install_mongo3() {
+    #install
+	rpm -i mongodb-org-server-3.6.17-1.el7.x86_64.rpm
+	rpm -i mongodb-org-shell-3.6.17-1.el7.x86_64.rpm
+	PATH=$PATH:/usr/bin; export PATH
+
+	#disable selinux
+	sed -i 's/^SELINUX=.*/SELINUX=disabled/' /etc/sysconfig/selinux
+	sed -i 's/^SELINUX=.*/SELINUX=disabled/' /etc/selinux/config
+	setenforce 0
+
+	#kernel settings
+	if [[ -f /sys/kernel/mm/transparent_hugepage/enabled ]];then
+		echo never > /sys/kernel/mm/transparent_hugepage/enabled
+	fi
+	if [[ -f /sys/kernel/mm/transparent_hugepage/defrag ]];then
+		echo never > /sys/kernel/mm/transparent_hugepage/defrag
+	fi
+
+	#configure
+	sed -i 's/\(bindIp\)/#\1/' /etc/mongod.conf
+	
+}
+
+yum install wget -y
+fqdn="${dnsNamePrefix}.${location}.cloudapp.azure.com"
+echo "Generating ssl certificate"
+openssl req -newkey rsa:2048 -nodes -keyout /etc/key.pem -x509 -days 365 -out /etc/certificate.pem -subj "/CN=$fqdn"
```

Comment:
as you have in the other scripts - remove the hardcoded password and use user supplied parameters